sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateAsyncClientTest.java

```diff
@@ -0,0 +1,1035 @@
+            PollerFlux<CertificateOperation, KeyVaultCertificateWithPolicy> certPoller =
+                certificateAsyncClient.beginCreateCertificate(certificateName, CertificatePolicy.getDefault());
+
+            AsyncPollResponse<CertificateOperation, KeyVaultCertificateWithPolicy> lastResponse =
+                certPoller.takeUntil(apr -> apr.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED)
+                    .blockLast();
+
+            StepVerifier.create(certificateAsyncClient.deleteCertificateOperation(certificateName))
+                    .assertNext(certificateOperation -> {
+                        assertEquals("completed", certificateOperation.getStatus());
+                    }).verifyComplete();
+
+
+            StepVerifier.create(certificateAsyncClient.deleteCertificateOperation(certificateName))
+                    .verifyErrorSatisfies(e -> {
+                        assertRestException(e, ResourceNotFoundException.class, HttpURLConnection.HTTP_NOT_FOUND);
+                    });
+        });
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void getCertificatePolicy(HttpClient httpClient, CertificateServiceVersion serviceVersion) {
+        createCertificateAsyncClient(httpClient, serviceVersion);
+
+        getCertificatePolicyRunner((certificateName) -> {
+            PollerFlux<CertificateOperation, KeyVaultCertificateWithPolicy> certPoller =
+                certificateAsyncClient.beginCreateCertificate(certificateName, setupPolicy());
+
+            AsyncPollResponse<CertificateOperation, KeyVaultCertificateWithPolicy> lastResponse =
+                certPoller.takeUntil(apr -> apr.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED)
+                    .blockLast();
+
+            StepVerifier.create(lastResponse.getFinalResult())
+                    .assertNext(certificate -> {
+                        validatePolicy(setupPolicy(), certificate.getPolicy());
+                    }).verifyComplete();
+        });
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void updateCertificatePolicy(HttpClient httpClient, CertificateServiceVersion serviceVersion) {
+        createCertificateAsyncClient(httpClient, serviceVersion);
+
+        updateCertificatePolicyRunner((certificateName) -> {
+            PollerFlux<CertificateOperation, KeyVaultCertificateWithPolicy> certPoller =
+                certificateAsyncClient.beginCreateCertificate(certificateName, setupPolicy());
+
+            AsyncPollResponse<CertificateOperation, KeyVaultCertificateWithPolicy> lastResponse =
+                certPoller.takeUntil(apr -> apr.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED)
+                    .blockLast();
+
+            KeyVaultCertificateWithPolicy certificate = lastResponse.getFinalResult().block();
+
```

Comment:
We should at least make one change to the policy to ensure it got updated like we do in the sync tests.
```suggestion
            KeyVaultCertificateWithPolicy certificate = lastResponse.getFinalResult().block();

            certificate.getPolicy().setExportable(false);
```